app/scripts/lib/setupSentry.js

```diff
@@ -23,53 +23,99 @@ export const ERROR_URL_ALLOWLIST = {
   SEGMENT: 'segment.io',
 };
 
-// This describes the subset of Redux state attached to errors sent to Sentry
-// These properties have some potential to be useful for debugging, and they do
-// not contain any identifiable information.
-export const SENTRY_STATE = {
-  gas: true,
-  history: true,
-  metamask: {
+// This describes the subset of background controller state attached to errors
+// sent to Sentry These properties have some potential to be useful for
+// debugging, and they do not contain any identifiable information.
+export const SENTRY_BACKGROUND_STATE = {
+  AccountTracker: {
+    currentBlockGasLimit: true,
+  },
+  AlertController: {
     alertEnabledness: true,
-    completedOnboarding: true,
+  },
+  AppStateController: {
     connectedStatusPopoverHasBeenShown: true,
+    defaultHomeActiveTabName: true,
+  },
+  CurrencyController: {
     conversionDate: true,
     conversionRate: true,
-    currentBlockGasLimit: true,
     currentCurrency: true,
-    currentLocale: true,
-    customNonceValue: true,
-    defaultHomeActiveTabName: true,
+    nativeCurrency: true,
+  },
+  DecryptMessageController: {
+    unapprovedDecryptMsgCount: true,
+  },
+  DesktopController: {
     desktopEnabled: true,
-    featureFlags: true,
-    firstTimeFlowType: true,
-    forgottenPassword: true,
+  },
+  EncryptionPublicKeyController: {
+    unapprovedEncryptionPublicKeyMsgCount: true,
+  },
+  IncomingTransactionsController: {
     incomingTxLastFetchedBlockByChainId: true,
-    ipfsGateway: true,
-    isAccountMenuOpen: true,
-    isInitialized: true,
+  },
+  KeyringController: {
     isUnlocked: true,
+  },
+  MetaMetricsController: {
     metaMetricsId: true,
-    nativeCurrency: true,
+    participateInMetaMetrics: true,
+  },
+  NetworkController: {
     networkId: true,
     networkStatus: true,
-    nextNonce: true,
-    participateInMetaMetrics: true,
-    preferences: true,
     providerConfig: {
       nickname: true,
       ticker: true,
       type: true,
     },
+  },
+  OnboardingController: {
+    completedOnboarding: true,
+    firstTimeFlowType: true,
     seedPhraseBackedUp: true,
-    unapprovedDecryptMsgCount: true,
-    unapprovedEncryptionPublicKeyMsgCount: true,
-    unapprovedMsgCount: true,
-    unapprovedPersonalMsgCount: true,
-    unapprovedTypedMessagesCount: true,
+  },
+  PreferencesController: {
+    currentLocale: true,
+    featureFlags: true,
+    forgottenPassword: true,
+    ipfsGateway: true,
+    preferences: true,
     useBlockie: true,
     useNonceField: true,
     usePhishDetect: true,
+  },
+  SignatureController: {
+    unapprovedMsgCount: true,
+    unapprovedPersonalMsgCount: true,
+    unapprovedTypedMessagesCount: true,
+  },
+};
+
+const flattenedBackgroundStateMask = Object.values(
+  SENTRY_BACKGROUND_STATE,
+).reduce((partialBackgroundState, controllerState) => {
```

Comment:
This reduce iterates over the controller objects within the `SENTRY_BACKGROUND_STATE` object defined above. On the first iteration, `partialBackgroundState` is an empty object and `controllerState` will be an object pointed at by one of the controller properties above. All the properties in that object get spread into the empty object and this is returned. On each successive iteration of the reduce, each of the controllers objects get spread onto the object that was returned in the previous iteration. The result is an object without nested objects, and every property is a property of a controller's state, each points to the value `true`